src/views/HomeView.tsx

```diff
@@ -9,27 +9,29 @@ import {
 } from '@geist-ui/react';
 import { DocumentHead, Container } from 'components';
 import { datasets } from 'mock/datasets';
-import { searchData } from 'mock/search';
 
 export const HomeView = () => {
     const [modalState, setModalState] = useState(false);
     const [path, setPath] = useState('');
-
-    const allOptions = searchData.map(({ name, path }) => {
-        return { label: name, value: name };
-    });
-
-    const [options, setOptions] = useState();
+    const [options, setOptions] = useState([]);
     const [searching, setSearching] = useState(false);
     const timer = useRef();
 
+    const allOptions = [];
+    datasets.forEach((dataset) => {
+        dataset.data.forEach((item) => {
+            const { name: value, path } = item;
+            allOptions.push({ value, label: value, path });
+        });
+    });
+
```

Comment:
When I'm working in React I like to go in a functional style as it is good for performance, no worries about performance here though :) but it is good practice.  Let me know what you think about using the map method.

 ```
const allOptions = datasets.map((dataset) => {
        return dataset.data.map((item) => {
            const { name: value, path } = item;
            return { value, label: value, path };
        });
    });
```
and remember in search handler to flatten the array above as it is nested `allOptions.flat().filter...`

Reply:
I agree. It is a good practice..